src/mpid/ch4/shm/posix/posix_am.h

```diff
@@ -11,6 +11,11 @@
 #include "posix_eager.h"
 #include "mpidu_genq.h"
 
+MPL_STATIC_INLINE_PREFIX size_t MPIDI_POSIX_am_eager_limit(void)
+{
+    return MPIDI_POSIX_eager_payload_limit() - MAX_ALIGNMENT;
```

Comment:
Why not discount `MAX_ALIGNMENT` inside `MPIDI_POSIX_eager_payload_limit()`?

Reply:
The `MPIDI_POSIX_eager_payload_limit` is provided by the individual eager modules. Since we are not maintaining all those modules, it just simpler to discount for `MAX_ALIGNMENT` at POSIX level.